src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1667 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+
+    status = app_trajectory_planner_generateConstantInterpolationVelocityTrajectory(
+        path_parameters, &velocity_trajectory, 0.010);
+    EXPECT_EQ(status, OK);
+
+    std::vector<Vector> directions =
+        getDirectionVectorsFromPositionTrajectory(&position_trajectory);
+    const float segment_length           = sqrt(8.0) / (path_parameters.num_segments - 1);
+    const float initial_segment_duration = (2 * segment_length) / (1);
+
+    for (unsigned int i = 0; i < velocity_trajectory.path_parameters.num_segments - 1;
+         i++)
+    {
+        EXPECT_FLOAT_EQ(velocity_trajectory.trajectory_elements[i].angular_velocity,
+                        position_trajectory.trajectory_elements[i].angular_speed);
+        EXPECT_FLOAT_EQ(
+            velocity_trajectory.trajectory_elements[i].linear_velocity.x,
+            position_trajectory.trajectory_elements[i].linear_speed * directions[i].x());
+        EXPECT_FLOAT_EQ(
+            velocity_trajectory.trajectory_elements[i].linear_velocity.y,
+            position_trajectory.trajectory_elements[i].linear_speed * directions[i].y());
+        EXPECT_NEAR(velocity_trajectory.trajectory_elements[i].time,
+                    position_trajectory.trajectory_elements[i].time, 0.00001);
+    }
+    EXPECT_FLOAT_EQ(
+        velocity_trajectory
+            .trajectory_elements[velocity_trajectory.path_parameters.num_segments - 1]
+            .angular_velocity,
+        0);
+}
+
+// TEST_F(
+//        TrajectoryPlannerTest,
+//        test_copy_position_trajectory_element)
+//{
+//
+// PositionTrajectory_t from_trajectory;
+// PositionTrajectoryElement_t
+//        from_trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+// from_trajectory.trajectory_elements = from_trajectory_elements;
+//
+// PositionTrajectory_t to_trajectory;
+// PositionTrajectoryElement_t
+// to_trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+// to_trajectory.trajectory_elements = to_trajectory_elements;
+//
+// from_trajectory.trajectory_elements[12].time = 1;
+// from_trajectory.trajectory_elements[12].linear_speed = 2;
+// from_trajectory.trajectory_elements[12].angular_speed = 3;
+// from_trajectory.trajectory_elements[12].orientation = 4;
+// Vector2d_t position = { .x = 5, .y = 6};
+// from_trajectory.trajectory_elements[12].position = position;
+//
+//
+// app_trajectory_planner_copyPositionTrajectoryElement(&to_trajectory.trajectory_elements[12],
+// &from_trajectory.trajectory_elements[12]);
+//
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].time, 1);
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].linear_speed, 2);
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].angular_speed, 3);
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].orientation, 4);
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].position.x, 5);
+// EXPECT_FLOAT_EQ(to_trajectory.trajectory_elements[12].position.y, 6);
+//}
```

Comment:
Well this is one way to make sure the tests pass......

Reply:
Outdated.